src/database/database.h

```diff
@@ -277,20 +279,28 @@ class DBTransactionGuard {
 		DBTransactionGuard &operator=(DBTransactionGuard &&) = delete;
 
 		~DBTransactionGuard() noexcept {
-			try {
-				transaction_.commit();
-			} catch (const std::exception &exception) {
-				// Error occurred while committing transaction
-				SPDLOG_ERROR("Error occurred while committing transaction", __FUNCTION__, exception.what());
+			if (!transaction_.isCommitted()) {
 				try {
 					transaction_.rollback();
 				} catch (const std::exception &exception) {
 					// Error occurred while rolling back transaction
-					SPDLOG_ERROR("Error occurred while rolling back transaction", __FUNCTION__, exception.what());
+					SPDLOG_ERROR("[{}] Error occurred while rolling back transaction, error: {}", __FUNCTION__, exception.what());
 				}
 			}
 		}
 
+		bool execute(std::function<void()> toBeExecuted) {
+			try {
+				toBeExecuted();
+				transaction_.commit();
+				return true;
+			} catch (const std::exception &exception) {
+				SPDLOG_ERROR("[{}] Error occurred while committing transaction, erro: {}", __FUNCTION__, exception.what());
+				throw;
+			}
+			return false;
+		}
```

Comment:
Nice change, it now does what it's supposed to do. I'd improve a bit to make it more consistent, moving begin and rollback here. This way you can call execute multiple times.
```
		static bool executeWithinTransaction(std::function<void()> toBeExecuted) {
			try {
        DBTransaction transaction;
        transaction_.begin();
				toBeExecuted();
				transaction_.commit();
				return true;
			} catch (const std::exception &exception) {
				SPDLOG_ERROR("[{}] Error occurred while committing transaction, erro: {}", __FUNCTION__, exception.what());
				transaction_.rollback();
			  return false;
			}
		}
```
No need to try catch the rollback, if it fails it fails and should bubble up (db issues).

Also, you can either re-throw an exception or return false, depending on what you wanna achieve. You cannot do both, one will always be ignored.

I'd change the name to 'executeWithinTransaction' for clarity. This could be a static function as well, so you could do `DBGuadd::executeWithinTransaction` directly;

For simplicity, it could even be a metthod of Transaction itself, then we could simplify removing the guard class.